packages/components/src/Filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { Dropdown, DropdownToggle, Level, DropdownProps } from '@patternfly/react-core';
+import FilterInput from './FilterInput';
+import './filter-dropdown.scss';
+
+export interface filterCategoriesProps {
+  filterCategories: [
+    {
+      title: string;
+      type: string;
+      urlParam: string;
+      values: [
+        {
+          label: React.ReactNode;
+          value: string;
+        }
+      ];
+    }
+  ];
+}
+
+export interface FilterDropdownProps extends Omit<DropdownProps, 'onSelect'> {
+  addFilter?: (filterName?: string, selectedValue?: string | number, type?: string) => void;
+  removeFilter?: (filterName?: string, selectedValue?: string | number) => void;
+  hideCategories?: [string];
+  isOpen: boolean;
```

Comment:
This attribute was originally handled through state, not props. Can you change it back, please? I think we have misunderstood ourselves in my previous review. I thought that you were asking about the internal `onToggle` function which was unnecessary and the state set function could have been called directly in the JSX part of the file.

Reply:
I don't get it, it just seems redundant to have setState when it is going to be the exact same value

Reply:
Well the `isOpen` variable cannot be migrated just like to be suddenly an outside prop. That is the main issue.

It was not set to be the exact same value but it was changed to the value sent by the toggle. We were only passing it along to the state The original code is as follows
```js
  onToggle = (isOpen /** the value is set by the PF component */) => {
    this.setState({
      isOpen,
    });
  };
```